src/main/java/org/balanpy/HigieneController.java

```diff
@@ -1,12 +1,219 @@
 package org.balanpy;
 
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextArea;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
 import javafx.scene.input.MouseEvent;
 
 public class HigieneController {
 
-	public void HandleDone(MouseEvent event) throws Exception {
+	private static final Path MASCOTAS_PATH = Paths.get(Configuracion.getRootDir(), "mascotas.json");
+
+	ArrayList<MascotaImpl> mascotas;
+
+	@FXML
+	private Button configHigiene;
+
+	@FXML
+	private Button pdfHigiene;
+
+	@FXML
+	private ImageView petImage0;
+
+	@FXML
+	private ImageView petImage1;
+
+	@FXML
+	private ImageView petImage2;
+
+	@FXML
+	private Label petName0;
+
+	@FXML
+	private Label petName1;
+
+	@FXML
+	private Label petName2;
+
+	@FXML
+	private TextField ultimoBano;
+
+	@FXML
+	private TextField ultimoCepillado;
+
+	@FXML
+	private TextArea productosNR;
+
+	@FXML
+	private TextArea ultimaDesp;
+
+	public HigieneController() {
+
+		try {
+			ObjectMapper om = new ObjectMapper();
+
+			mascotas = om.readValue(MASCOTAS_PATH.toFile(), new TypeReference<ArrayList<MascotaImpl>>() {
+
+			});
+
+		} catch (IOException e) {
+
+			System.out.println("Failed to load pet data. " + e.getMessage());
+
+			mascotas = new ArrayList<MascotaImpl>();
+
+		}
+
+	}
+
+	public void selectProfile(int id) throws Exception {
+
+		ultimoBano.setText(ultimoBanoToString(mascotas.get(id)));
+
+		ultimoCepillado.setText(ultimoCepilladoToString(mascotas.get(id)));
+
+		ultimaDesp.setText(desparasitadoToString(mascotas.get(id)));
+
+		productosNR.setText(productosToString(mascotas.get(id)));
```

Comment:
En lugar de llamar 4 veces a `mascotas.get(id)`, podemos crear una variable auxiliar.
```suggestion
		Mascota mascota = mascotas.get(id);
		ultimoBano.setText(ultimoBanoToString(mascota));

		ultimoCepillado.setText(ultimoCepilladoToString(mascota));

		ultimaDesp.setText(desparasitadoToString(mascota));

		productosNR.setText(productosToString(mascota));
```